From 0ba45ff04fb38f0041df355c0b14022ee2d0e327 Mon Sep 17 00:00:00 2001
From: ejgoldik <148076610+ejgoldik@users.noreply.github.com>
Date: Thu, 4 Sep 2025 16:44:40 +0300
Subject: [PATCH 54/80] d4xx-compat: relax start stream TO (#295)

* Relax Start stream TO
* Use only ds5_[raw]_<read/write> wrapper and not regmap_raw_<read/write> directly
---
 drivers/media/i2c/d4xx-compat/d4xx.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/media/i2c/d4xx-compat/d4xx.c b/drivers/media/i2c/d4xx-compat/d4xx.c
index 395ad192..54d132b2 100644
--- a/drivers/media/i2c/d4xx-compat/d4xx.c
+++ b/drivers/media/i2c/d4xx-compat/d4xx.c
@@ -193,7 +193,7 @@ enum ds5_mux_pad {
 #define DFU_WAIT_RET_LEN 6
 
 #define DS5_START_POLL_TIME	10
-#define DS5_START_MAX_TIME	1000
+#define DS5_START_MAX_TIME	2000
 #define DS5_START_MAX_COUNT	(DS5_START_MAX_TIME / DS5_START_POLL_TIME)
 
 /* DFU definition section */
@@ -2449,7 +2449,7 @@ static int ds5_get_hwmc(struct ds5 *state, unsigned char *data,
 		}
 	}
 
-	ret = regmap_raw_read(state->regmap, DS5_HWMC_RESP_LEN,
+	ret = ds5_raw_read(state, DS5_HWMC_RESP_LEN,
 			&tmp_len, sizeof(tmp_len)); /* Read response length */
 	if (ret)
 		return -EBADMSG;
@@ -2919,7 +2919,7 @@ static int ds5_get_calibration_data(struct ds5 *state, enum table_id id,
 	}
 
 	// get table length from fw
-	ret = regmap_raw_read(state->regmap, DS5_HWMC_RESP_LEN,
+	ret = ds5_raw_read(state, DS5_HWMC_RESP_LEN,
 			&table_length, sizeof(table_length)); /* Read response length */
 
 	// read table
@@ -2956,7 +2956,7 @@ static int ds5_gvd(struct ds5 *state, unsigned char *data)
 		return status;
 	}
 
-	ret = regmap_raw_read(state->regmap, DS5_HWMC_RESP_LEN, &length, sizeof(length)); /* Read response length */
+	ret = ds5_raw_read(state, DS5_HWMC_RESP_LEN, &length, sizeof(length)); /* Read response length */
 	ds5_raw_read_with_check(state, DS5_HWMC_DATA, data, length); /* Read response data */
 
 	return ret;
@@ -3077,18 +3077,18 @@ static int ds5_g_volatile_ctrl(struct v4l2_ctrl *ctrl)
 		//       2. send command
 		//       3. execute command
 		//       4. wait for completion
-		ret = regmap_raw_write(state->regmap, DS5_HWMC_DATA, /* Write command data */
+		ret = ds5_raw_write(state, DS5_HWMC_DATA, /* Write command data */
 				log_prepare, sizeof(log_prepare));
 		if (ret < 0)
 			return ret;
 
-		ret = regmap_raw_write(state->regmap, DS5_HWMC_EXEC,
+		ret = ds5_raw_write(state, DS5_HWMC_EXEC,
 				&execute_cmd, sizeof(execute_cmd)); /* execute cmd */
 		if (ret < 0)
 			return ret;
 
 		for (i = 0; i < DS5_MAX_LOG_POLL; i++) {
-			ret = regmap_raw_read(state->regmap, DS5_HWMC_STATUS,
+			ret = ds5_raw_read(state, DS5_HWMC_STATUS,
 					&data, sizeof(data));
 			dev_dbg(&state->client->dev, "%s(): log ready 0x%x\n",
 				 __func__, data);
@@ -3102,7 +3102,7 @@ static int ds5_g_volatile_ctrl(struct v4l2_ctrl *ctrl)
 //		if (i == DS5_MAX_LOG_POLL)
 //			return -ETIMEDOUT;
 
-		ret = regmap_raw_read(state->regmap, DS5_HWMC_RESP_LEN, &data, sizeof(data)); /* Read response length */
+		ret = ds5_raw_read(state, DS5_HWMC_RESP_LEN, &data, sizeof(data)); /* Read response length */
 		dev_dbg(&state->client->dev, "%s(): log size 0x%x\n", __func__, data);
 		if (ret < 0)
 			return ret;
@@ -3110,7 +3110,7 @@ static int ds5_g_volatile_ctrl(struct v4l2_ctrl *ctrl)
 			return 0;
 		if (data > 1024)
 			return -ENOBUFS;
-		ret = regmap_raw_read(state->regmap, DS5_HWMC_DATA,
+		ret = ds5_raw_read(state, DS5_HWMC_DATA,
 				ctrl->p_new.p_u8, data);
 		break;
 	case DS5_CAMERA_DEPTH_CALIBRATION_TABLE_GET:
-- 
2.25.1

